android-tests/README.md

```diff
@@ -0,0 +1,40 @@
+This is an Android project designed to run Chicory tests on device.
+
+Since Chicory uses maven and Android doesn't have official maven
+support, we cannot make Android testing part of the main build.
+
+Instead, we use this project where it uses the outputs of the
+main maven build to run those tests as Android instrumentation tests.
+
+Inside the `device-tests` folder, there is an Android library project
+setup. It doesn't have any code, except for declaring product flavors
+for each main chicory project and setting up its dependencies to run
+tests.
+
+Inside the root `build.gradle.kts` file, we setup a task that creates a
+maven repository from the outputs of the main project.
+
+The dependencies between this project and the maven project are setup properly
+such that, if the code in the main maven project changes, this Android project
+will recompile the repository and run up-to-date tests.
+
+To avoid re-building the main project (e.g. in CI), you can also pass
+`CHICORY_REPO` environment variable, in which case, this Android project will
+re-use its output instead of recompiling the main project.
+(It won't make any attempt to compile their tests either)
+```
+mvn deploy -DaltDeploymentRepository=local-repo::default::file:./local-repo -DskipTests
+cd android-tests && CHICORY_REPO=../local-repo ./gradlew device-tests:connectedCheck
```

Comment:
I attempted to go through the instructions but I got stuck here:

```
Define a valid SDK location with an ANDROID_HOME
```

I don't have an Android SDK installed globally, I attempted to download a few packages, but it seems not enough, can I ask for a link to a guide on how to have a working environment? Preferably on the command line ...

Reply:
Sorry i missed this. I'll write some instructions for it.
The easiest way is to actually use android studio. We also have command line tools. Unfortunately, it is a [bunch of steps](https://github.com/ReactiveCircus/android-emulator-runner/blob/main/src/sdk-installer.ts#L7) and I don't think we have a clear documentation around it that I can link to.

I'll see how complex that is and worst case we can always have the studio instructions (sorry, Android development and Android studio are very interlocked)

Reply:
I think with homebrew you can install `android-platform-tools`, `android-sdk` and `android-commandlinetools` and get pretty much there. Then you'll have to setup a virtual device (e.g. I am using Medium Phone API 35). Indeed, the easiest way is to setup through Android Studio -- also installing the Android plugin on a plain IntelliJ will get you there.